File: web/src/api/interaction.ts
import request from './request'
import type { DanmakuItem } from '@/types'

// 点赞视频
export function likeVideo(video_id: number) {
  return request.post<any, void>('/interaction/video/like', { video_id })
}

// 取消点赞
export function unlikeVideo(video_id: number) {
  return request.post<any, void>('/interaction/video/unlike', { video_id })
}

// 收藏视频
export function favoriteVideo(video_id: number, folder_id: number = 0) {
  return request.post<any, void>('/interaction/video/favorite', { video_id, folder_id })
}

// 取消收藏
// folder_id 可选，0 或不传表示从所有收藏夹移除
export function unfavoriteVideo(video_id: number, folder_id: number = 0) {
  return request.post<any, void>('/interaction/video/unfavorite', { video_id, folder_id })
}

// 关注用户
// 后端 FollowUserReq 字段为 target_user_id（json:"target_user_id"）
export function followUser(target_user_id: string) {
  return request.post<any, void>('/interaction/follow', { target_user_id })
}

// 取消关注
export function unfollowUser(target_user_id: string) {
  return request.post<any, void>('/interaction/unfollow', { target_user_id })
}

// 获取弹幕
export function getDanmaku(video_id: number) {
  return request.get<any, DanmakuItem[]>('/interaction/video/danmaku', { params: { video_id } })
}

// 发送弹幕
export function sendDanmaku(data: {
  video_id: number
  content: string
  video_time: number
  color: string
  font_size: string
  mode: number // 0=滚动, 1=顶部, 2=底部
}) {
  return request.post<any, void>('/interaction/video/danmaku', data)
}

File: server/internal/routers/interaction.go
package routers

import (
	"fake_tiktok/internal/handler"

	"github.com/gin-gonic/gin"
)

type InteractionRouter struct{}

func (i *InteractionRouter) InitInteractionRouter(publicGroup, privateGroup *gin.RouterGroup, handlerGroup *handler.HandlerGroup) {
	// 公开路由：GET 弹幕列表（未登录用户也能查看弹幕）
	publicInteraction := publicGroup.Group("interaction/video")
	{
		publicInteraction.GET("danmaku", handlerGroup.InteractionHandler.GetDanmakuList)
	}

	// 私有路由：需要登录才能访问的操作
	interactionRouter := privateGroup.Group("interaction")
	interactionVideoRouter := interactionRouter.Group("video")
	{
		interactionVideoRouter.POST("like", handlerGroup.InteractionHandler.LikeVideo)
		interactionVideoRouter.POST("favorite", handlerGroup.InteractionHandler.FavoriteVideo)
		interactionVideoRouter.POST("unfavorite", handlerGroup.InteractionHandler.UnfavoriteVideo)
		interactionVideoRouter.POST("unlike", handlerGroup.InteractionHandler.UnlikeVideo)
		interactionVideoRouter.POST("danmaku", handlerGroup.InteractionHandler.SendDanmaku)
	}
	{
		interactionRouter.POST("follow", handlerGroup.InteractionHandler.FollowUser)
		interactionRouter.POST("unfollow", handlerGroup.InteractionHandler.UnfollowUser)
	}
}